src/firefly/js/ui/ValidationField.jsx

```diff
@@ -1,13 +1,19 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import {has} from 'lodash';
-
+import {has, get} from 'lodash';
 import {InputFieldView} from './InputFieldView.jsx';
 import {fieldGroupConnector} from './FieldGroupConnector.jsx';
 
 
 function onChange(ev, store, fireValueChange) {
     var value = ev.target.value;
+
+    //html5 validation, using validity property object. If it is marked invalid, then no need to sent it to validator.js
+    if(! get(ev, 'target.validity.valid')) {
```

Comment:
One more question. I thought I asked this before but must have forgot.  Why is this `if` block of code even necessary? It the validation flow now working right? I think it is fine for other types of input data, what if different about date?

Reply:
This if is checking if the input field is marked invalid by html5 validation, if so then no need to curry on with our validator, and simply return an invalid message back to the user.

Reply:
ok, makes sense. thanks.

Reply:
So maybe I need to understand more.  We have to have validity as the state of the field or we can validate a whole FieldGroup.  i guess I am going to have to really study this then to understand the impact.  Please don't merge until I can.

Reply:
I used this link for the validity property:
https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Form_validation

Reply:
That is not wrong at all, it is just we need to validation state to be in our store.  We cannot be dependent on UI for state.

Reply:
I see. So far I'm not using it at all, since I'm using a plain input field (with no type='date'). I was using the validity object when I was using the html5 calendar widget. So I can remove it.

Reply:
ok, i guess you should.  I would like it all to work right with the html calendar widget eventually, but I going to need a chunk of time to think though it which I can't do while I am at LSST2018 conf.

Reply:
Are you ok with the validateDate function in validator.js. If so then I will remove the code on the validationField.jsx (the html5 validity stuff), and curry on.

Reply:
That's fine.